api/core/v1beta2/machinedeployment_types.go

```diff
@@ -281,52 +276,61 @@ type MachineDeploymentSpec struct {
 	// +required
 	Template MachineTemplateSpec `json:"template"`
 
-	// strategy is the deployment strategy to use to replace existing machines with
-	// new ones.
-	// +optional
-	Strategy MachineDeploymentStrategy `json:"strategy,omitempty,omitzero"`
-
 	// machineNamingStrategy allows changing the naming pattern used when creating Machines.
 	// Note: InfraMachines & BootstrapConfigs will use the same name as the corresponding Machines.
 	// +optional
 	MachineNamingStrategy *MachineNamingStrategy `json:"machineNamingStrategy,omitempty"`
 
+	// remediation controls how unhealthy Machines are remediated.
+	// +optional
+	Remediation MachineDeploymentRemediationSpec `json:"remediation,omitempty,omitzero"`
+
+	// deletion contains configuration options for MachineDeployment deletion.
+	// +optional
+	Deletion MachineDeploymentDeletionSpec `json:"deletion,omitempty,omitzero"`
+
 	// paused indicates that the deployment is paused.
 	// +optional
 	Paused *bool `json:"paused,omitempty"`
 }
 
 // ANCHOR_END: MachineDeploymentSpec
 
-// ANCHOR: MachineDeploymentStrategy
-
-// MachineDeploymentStrategy describes how to replace existing machines
-// with new ones.
+// MachineDeploymentRolloutSpec defines the rollout behavior.
 // +kubebuilder:validation:MinProperties=1
-type MachineDeploymentStrategy struct {
-	// type of deployment. Allowed values are RollingUpdate and OnDelete.
-	// The default is RollingUpdate.
-	// +optional
-	Type MachineDeploymentStrategyType `json:"type,omitempty"`
-
-	// rollingUpdate is the rolling update config params. Present only if
-	// MachineDeploymentStrategyType = RollingUpdate.
+type MachineDeploymentRolloutSpec struct {
+	// after is a field to indicate a rollout should be performed
+	// after the specified time even if no changes have been made to the
+	// MachineDeployment.
+	// Example: In the YAML the time can be specified in the RFC3339 format.
+	// To specify the rolloutAfter target as March 9, 2023, at 9 am UTC
+	// use "2023-03-09T09:00:00Z".
 	// +optional
-	RollingUpdate MachineDeploymentStrategyRollingUpdate `json:"rollingUpdate,omitempty,omitzero"`
+	After *metav1.Time `json:"after,omitempty"`
 
-	// remediation controls the strategy of remediating unhealthy machines
-	// and how remediating operations should occur during the lifecycle of the dependant MachineSets.
+	// strategy specifies how to roll out control plane Machines.
 	// +optional
-	Remediation RemediationStrategy `json:"remediation,omitempty,omitzero"`
+	Strategy MachineDeploymentRolloutStrategy `json:"strategy,omitempty,omitzero"`
```

Comment:
Point from Joel from the issue
> rollout.strategy should be implemented now as a discriminated union, will need to check if that's possible given prior validations

I looked into the validation (webhook + OpenAPI), we had absolutely no validation for discriminated union in v1beta1. So I'm very hesitant to introduce it now, especially as part of this PR.